compiler/rustc_attr/src/builtin.rs

```diff
@@ -870,6 +870,23 @@ pub fn find_repr_attrs(sess: &Session, attr: &Attribute) -> Vec<ReprAttr> {
                         sym::simd => Some(ReprSimd),
                         sym::transparent => Some(ReprTransparent),
                         sym::no_niche => Some(ReprNoNiche),
+                        sym::align => {
+                            let mut err = struct_span_err!(
+                                diagnostic,
+                                item.span(),
+                                E0589,
+                                "invalid `repr(align)` attribute: `align` needs an argument"
+                            );
+                            err.span_suggestion(
+                                item.span(),
+                                "supply an argument here",
+                                "align(...)".to_string(),
+                                Applicability::HasPlaceholders,
+                            );
+                            err.emit();
```

Comment:
Because you have no conditional changes on the diagnostic, you can rely on chaining instead:

```suggestion
                            struct_span_err!(
                                diagnostic,
                                item.span(),
                                E0589,
                                "invalid `repr(align)` attribute: `align` needs an argument"
                            ).span_suggestion(
                                item.span(),
                                "supply an argument here",
                                "align(...)".to_string(),
                                Applicability::HasPlaceholders,
                            ).emit();
```

(Might need to run `./x.py fmt` to fix the formatting here)